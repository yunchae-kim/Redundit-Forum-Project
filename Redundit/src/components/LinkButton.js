import "../style/components/LinkButton.css";
import { Link } from "react-router-dom";

const LinkButton = ({ url, text, groupId }) => (
  <Link to={{ pathname: url, state: { groupId } }}>
    <button type="button" className="link-button">{text}</button>
  </Link>
);

export default LinkButton;
