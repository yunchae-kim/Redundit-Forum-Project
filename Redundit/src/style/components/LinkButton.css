.link-button {
  width: 280px;
  height: 40px;
  margin-bottom: 10px;
  border: none;
  border-radius: 2rem;
  background-color: #188bd8;
  color: white;
  font-size: large;
  font-weight:600;
}