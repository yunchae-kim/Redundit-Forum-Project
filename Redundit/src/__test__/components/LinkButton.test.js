import {
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import LinkButton from "../../components/LinkButton";

test("test should LinkButton match snapshot", async () => {
  const tree = renderer.create(
    <BrowserRouter>
      <LinkButton url="test" text="123" groupId={1} />
    </BrowserRouter>,
  ).toJSON();
  await waitFor(async () => {
    expect(tree).toMatchSnapshot();
  });
});
