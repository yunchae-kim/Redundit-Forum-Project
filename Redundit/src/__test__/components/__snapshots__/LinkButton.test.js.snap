// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should LinkButton match snapshot 1`] = `
<a
  href="test"
  onClick={[Function]}
>
  <button
    className="link-button"
    type="button"
  >
    123
  </button>
</a>
`;
